packages/cosmo/package.py

```diff
@@ -245,6 +245,11 @@ def setup_build_environment(self, env):
         else:
             env.set('GRIBAPII', '')
 
+        # libaec
+        if 'libaec' in self.spec:
```

Comment:
did you test this ? 
I was wondering how would that work because libaec is not a variant of cosmo package. 
There is only a +aec (not +libaec) in eccodes.
So I assuming that self.spec contains only the spec of cosmo and not the dependencies, so I guess you would need to 
(in pseudo code)
``` 
if +eccodes and +aec self.eccodes.spec 
```

Reply:
Yes, I did test it.
Well, If you concretize the spec, all the dependencies are specified. And with +aec, libaec is a dependency. Thus ^libaec is now in the spec.

That's what python prints for `self.spec`, notice the ^libaec in there :
``` cosmo@apn_5.09a.mch1.2.p1%pgi@20.4~claw+cppdycore~debug+dycoretest+eccodes~gt1~oasis+parallel~pollen~production~serialize~set_version~verbose~zlib_ng cosmo_target=gpu cuda_arch=70 real_type=double slave=tsa slurm_account=- slurm_bin=srun slurm_constraint=- slurm_nodes=1 slurm_opt_account=- slurm_opt_constraint=- slurm_opt_nodes=-n slurm_opt_partition=-p slurm_partition=debug arch=linux-rhel7-skylake_avx512 ^boost@1.67.0%gcc@8.3.0+atomic+chrono~clanglibcpp~container~context~coroutine+date_time~debug+exception~fiber+filesystem+graph~icu+iostreams+locale+log+math~mpi+multithreaded~numpy~pic+program_options~python+random+regex+serialization+shared+signals~singlethreaded+system~taggedlayout+test+thread+timer~versionedlayout+wave cxxstd=98 patches=7f95f95be9645eb7f10a7222173c8549501aebbe1db12b955442a7554dc59f3e visibility=hidden arch=linux-rhel7-skylake_avx512 ^bzip2@1.0.6%gcc@8.3.0~debug~pic+shared arch=linux-rhel7-skylake_avx512 ^cmake@3.14.5%gcc@8.3.0~doc+ncurses+openssl+ownlibs~qt build_type=Release patches=1c540040c7e203dd8e27aa20345ecb07fe06570d56410a24a266ae570b1c4c39 arch=linux-rhel7-skylake_avx512 ^cosmo-dycore@apn_5.09a.mch1.2.p1%gcc@8.3.0+build_tests+cuda~gt1~ipo~pmeters~production build_type=Release cuda_arch=70 data_path=/scratch/jenkins/data/cosmo/ real_type=double slave=tsa slurm_account=- slurm_bin=srun slurm_constraint=- slurm_gpu=--gres=gpu:{0} slurm_nodes={0} slurm_opt_account=- slurm_opt_constraint=- slurm_opt_nodes=-n slurm_opt_partition=-p slurm_partition=debug arch=linux-rhel7-skylake_avx512 ^cosmo-eccodes-definitions@2.19.0.7%pgi@20.4 arch=linux-rhel7-skylake_avx512 ^cuda@10.1.243%gcc@8.3.0~dev arch=linux-rhel7-skylake_avx512 ^eccodes@2.19.0%pgi@20.4+aec~build_shared_libs+examples+fortran~ipo~memfs~netcdf~openmp~png~pthreads~python+test build_type=RelWithDebInfo jp2k=jasper arch=linux-rhel7-skylake_avx512 ^jasper@1.900.1%gcc@8.3.0+jpeg~opengl+shared build_type=Release patches=db104400a2e72f610b8fa4d061a32282254819c70b024ef1cf99fef64aca67e3 arch=linux-rhel7-skylake_avx512 ^libaec@1.0.5%pgi@20.4~ipo build_type=RelWithDebInfo arch=linux-rhel7-skylake_avx512 ^libgrib1@22-01-2020%pgi@20.4 arch=linux-rhel7-skylake_avx512 ^netcdf-c@4.7.0%pgi@20.4~dap~fsync~hdf4~jna+mpi~parallel-netcdf+pic+shared patches=10a1c3f7fa05e2c82457482e272bbe04d66d0047b237ad0a73e87d63d848b16c arch=linux-rhel7-skylake_avx512 ^netcdf-fortran@4.4.5%pgi@20.4~doc+pic+shared patches=55fef65a4652fc5b8114db25bc558c762d9d4d287377a7db0ea2a2937d625e44,ef26f0860c0d258af7ba28780a981545a8a1c053dbdabe2d656d6e40ce1c7c06 arch=linux-rhel7-skylake_avx512 ^openmpi@4.0.2%pgi@20.4~atomics+cuda~cxx~cxx_exceptions+gpfs~internal-hwloc~java~legacylaunchers~lustre~memchecker~pmi~pmix~singularity~sqlite3+static~thread_multiple+vt+wrapper-rpath fabrics=none patches=073477a76bba780c67c36e959cd3ee6910743e2735c7e76850ffba6791d498e4,60ce20bc14d98c572ef7883b9fcd254c3f232c2f3a13377480f96466169ac4c8 schedulers=none arch=linux-rhel7-skylake_avx512 ^perl@5.16.3%pgi@20.4+cpanm+shared+threads patches=0eac10ed90aeb0459ad8851f88081d439a4e41978e586ec743069e8b059370ac,3bbd7d6f9933d80b9571533867b444c6f8f5a1ba0575bfba1fba4db9d885a71a arch=linux-rhel7-skylake_avx512 ^serialbox@2.6.0%gcc@8.3.0+boost_cmake+boost_sys_paths+exp_filesystem+fortran+ftg~ipo~netcdf+no_pckg_registery+python+sdb+shared~testing_fortran~testing_gridtools~testing_stella build_type=Release arch=linux-rhel7-skylake_avx512 ^slurm@19.05.05%gcc@8.3.0~gtk~hdf5~hwloc~mariadb~pmix+readline~restd sysconfdir=PREFIX/etc arch=linux-rhel7-skylake_avx512 ^zlib@1.2.11%gcc@8.3.0+optimize+pic+shared arch=linux-rhel7-skylake_avx512```

Reply:
alright, thanks, make sense then, yes